benches/lib.rs

```diff
@@ -334,3 +345,85 @@ fn to_string_plus_op_test() {
     let datetime: &str = &(DATE.to_string() + T + TIME);
     assert_eq!(String::from(DATETIME), datetime);
 }
+
+// ===== MACRO TESTS =====
+
+/// https://crates.io/crates/concat-in-place
+#[bench]
+fn concat_in_place_macro(b: &mut Bencher) {
+    b.iter(|| {
+        let mut url = String::new();
+        let datetime = concat_in_place::strcat!(&mut url, DATE T TIME);
+        test::black_box(datetime);
+    });
+}
+
+#[test]
+fn concat_in_place_macro_test() {
+    let mut url = String::new();
+    let datetime = concat_in_place::strcat!(&mut url, DATE T TIME);
+    assert_eq!(&String::from(DATETIME), datetime);
+}
+
+/// https://crates.io/crates/string_concat
+#[bench]
+fn string_concat_macro(b: &mut Bencher) {
+    b.iter(|| {
+        let datetime = &string_concat::string_concat!(DATE, T, TIME);
+        test::black_box(datetime);
+    });
+}
+
+#[test]
+fn string_concat_macro_test() {
+    let datetime = &string_concat::string_concat!(DATE, T, TIME);
+    assert_eq!(&String::from(DATETIME), datetime);
+}
+
+/// https://crates.io/crates/concat_strs
+/// This macro breaks RustAnalyzer (https://github.com/rust-analyzer/rust-analyzer/issues/6835)
+#[bench]
+fn concat_strs_macro(b: &mut Bencher) {
+    b.iter(|| {
+        let datetime = &concat_strs!(DATE, T, TIME);
+        test::black_box(datetime);
+    });
+}
+
+#[test]
+fn concat_strs_macro_test() {
+    let datetime = &concat_strs!(DATE, T, TIME);
+    assert_eq!(&String::from(DATETIME), datetime);
+}
+
+/// https://crates.io/crates/concat-string
+#[bench]
+fn concat_string_macro(b: &mut Bencher) {
+    b.iter(|| {
+        let datetime = concat_string!(DATE, T, TIME);
+        test::black_box(datetime);
+    });
+}
+
+#[test]
+fn concat_string_macro_test() {
+    let datetime = &concat_string!(DATE, T, TIME);
+    assert_eq!(&String::from(DATETIME), datetime);
+}
+
+/// https://crates.io/crates/joinery
+#[bench]
+fn joinery(b: &mut Bencher) {
+    let vec = vec![DATE, T, TIME];
+    b.iter(|| {
+        let datetime = &vec.iter().join_with("").to_string();
```

Comment:
Recommend instead, to avoid penalties:

Version 1:

```rust
let parts = [DATE, T, TIME];
...
let datetime = parts.join_concat().to_string();
```

Version 2:

```rust
let parts = [DATE, TIME];
...
let datetime = parts.join_with(T).to_string();
```

Reply:
Used Version 1 to keep it consistent with the other benches. It dropped from 57 to 46ns. @Lucretiel , please, double-check if you have a minute.